### PR TITLE
Email confirmation feature

### LINKED ISSUE #221: Cannot create an account when require emails confirmation is set to YES
**Describe the bug**
When `Require Emails Confirmation` is set to `Yes` and I try to create an account, a confirmation letter is not sent, the `This account isn't confirmed. Verify and try again.` error pops up and sign up dropdown is loading.

**To Reproduce**
1. On the admin sidebar, tap `Stores`. Under `Settings` choose `Configuration`
2. In the panel on the left under `Customers` choose `Customer Configuration`
3. Expand the `Create New Account Options` section. 
4. Set `Require Emails Confirmation` to `Yes`
5. On the customer view, create a new account.
6. **Notice the text described above**

**Expected behavior**
The error should not pop up before the confirmation email is sent. The dropdown should not be loading.

**Screenshots**
Screen Shot 2019-07-12 at 11 32 39

### PR DESCRIPTION
**Pull request by dmitrijs-voronovs**
- Added `ConfirmAccountPage` component
- Integrated account confirmation page into router. Page can be accessed by address http://scandipwa.local/account/confirm/
- Added `ConfirmAccountMutation`
- Added `confirmaiton_required` field to Customer
- Updated dispatcher to support Account confirmation
- Fixed url parameter decoding to correctly decode special characters. Was used to decode email from get params on `ConfirmAccountPage`.
- Fixed `MyAccountOverlay`. Now it hides after creating account so user can see notification.
- Fixed #221

Depends on https://github.com/scandipwa/customer-graph-ql/pull/6

For enabling the feature:
1. Inside M2 admin navigate to `Stores > Configuration > Customers > Customer Configuration > Create New Account Options`
2. Set `Require Emails Confirmation` to `Yes`
3. For getting correct links, set `Stores > Configuration > General > Web > Search Engine Optimization > Use Web Server Rewrites` to `Yes`

For testing:
1. Enable feature as described above
2. Create an Account
3. Go to `http://scandipwa.local:1080/maildev/#/` (integrated mail for testing purposes). Find `please confirm your account mail` and press the `Confirm Your Account` button
4. You will be redirected to new page. Input your password and press `Confirm` button
5. If password is correct, your account gets confirmed, you automatically get authorized and redirected to `my-account` page. Otherwise you are notified that something went wrong.
6. While user is logged in, `confirm account` page is not available for him. 

Screenshots:
- Mobile 
Selection_010
- Desktop
Selection_062

### CODE REVIEW

**Comment by alfredsgenkins** [comment]
Please decrease size of the heading on mobile :)

File: src/app/route/ConfirmAccountPage/ConfirmAccountPage.container.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    onFormError() {
+        this.setState({ isLoading: false });
+    }
+
+    _updateBreadcrumbs() {
+        const { updateBreadcrumbs } = this.props;
+        const breadcrumbs = [
+            {
+                url: '/accout/confirmAccount',
```
Comment (atravkovs) [inline src/app/route/ConfirmAccountPage/ConfirmAccountPage.container.js]: I would prefer `/account/` instead of `/accout/` here

**Comment by atravkovs** [comment]
@dmitrijs-voronovs Have you checked it working without webpack-dev-server (like prod version, that is without `- f docker-compose.frontend.yml`)? Usually adding new URL address, requires changes in `route717`...

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/route/ConfirmAccountPage/ConfirmAccountPage.component.js  3
- src/app/route/ConfirmAccountPage/ConfirmAccountPage.container.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ScandiPWA/base-theme/pullRequest?prid=4428273&bid=14988664)

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/route/ConfirmAccountPage/ConfirmAccountPage.component.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ScandiPWA/base-theme/pullRequest?prid=4428273&bid=14988664)

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/route/ConfirmAccountPage/ConfirmAccountPage.component.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ScandiPWA/base-theme/pullRequest?prid=4428273&bid=14988664)

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/route/ConfirmAccountPage/ConfirmAccountPage.component.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ScandiPWA/base-theme/pullRequest?prid=4428273&bid=14988664)

**Comment by dmitrijs-voronovs** [comment]
Fixed issues and merge conflicts.

----
Tested everything and found some bugs on BE (described there accordingly)
The only FE enhancement can be made is to show norification that user should confirm his account using link in the email.

Will fix BE and do FE enhancements as soon as possible.

**Comment by dmitrijs-voronovs** [comment]
> Fixed issues and merge conflicts.
> 
> Tested everything and found some bugs on BE (described there accordingly)
> The only FE enhancement can be made is to show norification that user should confirm his account using link in the email.
> 
> Will fix BE and do FE enhancements as soon as possible.

Fixed! Ready to merge!

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/route/ConfirmAccountPage/ConfirmAccountPage.component.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ScandiPWA/base-theme/pullRequest?prid=4428273&bid=14988664)

**Comment by dmitrijs-voronovs** [comment]
Ready!

**Pull request merged**